lib/internal/child_process.js

```diff
@@ -926,7 +929,8 @@ function _validateStdio(stdio, sync) {
       acc.push({
         type: 'wrap',
         wrapType: getHandleWrapType(handle),
-        handle: handle
+        handle: handle,
+        _stdio: stdio
```

Comment:
So … the code above (https://github.com/nodejs/node/pull/22892/files#diff-f45eb699237c2e38dc9b49b588933c11R923) reads like `stdio` could be:

- A native stream handle
- An object with a `handle` property which is a native stream handle
- An object with a `_handle` property which is a native stream handle

The path that is typically taken for e.g. `net.Socket`s would be the third one, I think. We should probably look into deprecating at least the first two, but … all that’s a different story and not really for this PR.

What matters is that `stdio` does not need to be any kind of particular object; most of the time, it’s a `net.Socket` or similar, but it could also just be `{ handle: someNativeHandle }`. Even though that’s a bad idea imo, the code above points to this being a possibility. This means that the `stream._stdio.emit()` call above might fail, because `emit()` might not be available?

I guess a simple solution would be to turn the `if (stream._stdio) {` conditional into `if (stream._stdio && typeof stream._stdio.emit === 'function') { `?

Reply:
I see.
I came up with a way when object is pushed. How about this?
Should it be judged when `emit()`?
```
const a = {
  type: 'wrap',
  wrapType: getHandleWrapType(handle),
  handle: handle
};

if (stdio.emit === 'function') {
  a._stdio = stdio;
}

acc.push(a);
```

Reply:
Or how about `if (stream._stdio instanceof EventEmitter)`?

Reply:
@koh110 I’m not sure. `instanceof EventEmitter` is probably fine, too?

I would prefer for the `_stdio` property to be present consistently, though – either it’s always there or always missing.

Reply:
@addaleax Yeah, you right. I completely agree. I was wrong.
> I would prefer for the _stdio property to be present consistently,

I feel that `if (stream._stdio && stream._stdio instanceof EventEmitter) {` is more meaningful than `if(stream._stdio && typeof stream._stdio.emit === 'function') {`. `EventEmitter` may be changed to `WritableStream`. What do you think about it?